sample/src/main/kotlin/com/slack/circuit/sample/petdetail/PetDetail.kt

```diff
@@ -121,35 +124,49 @@ class PetDetailScreenFactory @Inject constructor() : ScreenViewFactory {
   }
 }
 
-private fun petDetailUi() = ui<PetDetailScreen.State, Nothing> { state, _ -> RenderImpl(state) }
+private fun petDetailUi() = ui<PetDetailScreen.State, Nothing> { state, _ -> PetDetail(state) }
 
 @Composable
-private fun RenderImpl(state: PetDetailScreen.State) {
+internal fun PetDetail(state: PetDetailScreen.State) {
   Scaffold(modifier = Modifier.systemBarsPadding()) { padding ->
     when (state) {
       PetDetailScreen.State.Loading -> {
         Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
-          CircularProgressIndicator()
+          CircularProgressIndicator(modifier = Modifier.testTag("progress"))
+        }
+      }
+      PetDetailScreen.State.UnknownAnimal -> {
+        Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+          Text(
+            modifier = Modifier.testTag("unknown animal"),
```

Comment:
I think we should extract internal `const val` props for these and reuse them

Reply:
done